contrib/golang/filters/http/source/golang_filter.cc

```diff
@@ -897,6 +897,49 @@ CAPIStatus Filter::getStringValue(int id, GoString* value_str) {
   return CAPIStatus::CAPIOK;
 }
 
+CAPIStatus Filter::log(uint32_t level, absl::string_view message) {
+  Thread::LockGuard lock(mutex_);
+  if (has_destroyed_) {
+    ENVOY_LOG(debug, "golang filter has been destroyed");
+    return CAPIStatus::CAPIFilterIsDestroy;
+  }
+  auto& state = getProcessorState();
+  if (!state.isProcessingInGo()) {
```

Comment:
seems it's safe to log here? it only depends on the Filter is alive?

Reply:
> seems it's safe to log here? it only depends on the Filter is alive?

This API is only allowed to be used on the Go plugin side. It is intended to unify the logging prefix format for Go plugins, and also to avoid mixing logs with those of Go extension filters. prefix format such as:  `"[go_plugin_http][{}] {}", config_->pluginName(), message`.